### PR TITLE
Deploy/v8.4.0

### PR DESCRIPTION
**Pull request by rmothilal**
Added:
 - ALS-Oracle-Pathfinder
 - Transacrtion-Request-Service

### CODE REVIEW

File: account-lookup-service/requirements.yaml [hunk 1]
```diff
@@ -8,6 +8,10 @@ dependencies:
   version: 8.3.0
   repository: "file://./chart-admin"
   condition: account-lookup-service-admin.enabled
+- name: als-oracle-pathfinder
```
Comment (mdebarros) [inline account-lookup-service/requirements.yaml]: Ensure that you update the `account-lookup-service/chart.yaml` appVersion field to include the als-oracle-pathfinder

File: mojaloop/values.yaml [hunk 2]
```diff
@@ -2881,6 +2881,122 @@ account-lookup-service:
 
     affinity: {}
 
+  als-oracle-pathfinder:
+    enabled: false
```
Comment (mdebarros) [inline mojaloop/values.yaml]: Note that this flag will not work. I assume you have tested the deployment?

Please confirm if the als-oracle-pathfinder is being deployed even through `enabled: false` has been configured?

File: transaction-requests-service/values.yaml [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+    ## Set toleration for schedular
+    ## ref: https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/
+    tolerations: []
+
+# Declare variables to be passed into your templates.
+replicaCount: 1
+image:
+  repository: mojaloop/transaction-requests-service
+  tag: v8.3.0-snapshot
```
Comment (mdebarros) [inline transaction-requests-service/values.yaml]: This docker image does not seem to be published to the mojaloop docker repo?

Please check the license check. Also did you verify that this deploys correctly when testing the deployment?

File: update-charts-dep.sh [hunk 4]
```diff
@@ -41,6 +41,10 @@ echo "Updating Account Lookup Service..."
 helm dep up ./account-lookup-service
 checkCommandResult
 
+echo "Updating ALS Oracle Pathfinder..."
```
Comment (mdebarros) [inline update-charts-dep.sh]: You need to update the publish.sh script with a similar update.

File: update-charts-dep.sh [hunk 5]
```diff
@@ -73,6 +77,10 @@ echo "Updating quoting-service..."
 helm dep up ./quoting-service
 checkCommandResult
 
+echo "Updating transaction-request-service..."
```
Comment (mdebarros) [inline update-charts-dep.sh]: You need to update the publish.sh script with a similar update.

File: mojaloop/values.yaml (hunk 2 above)
Comment (rmothilal) [reply mojaloop/values.yaml]: @mdebarros it does not deploy when enable is set to false

File: transaction-requests-service/values.yaml (hunk 3 above)
Comment (rmothilal) [reply transaction-requests-service/values.yaml]: fixed

File: update-charts-dep.sh (hunk 4 above)
Comment (rmothilal) [reply update-charts-dep.sh]: added to package.sh

File: update-charts-dep.sh (hunk 5 above)
Comment (rmothilal) [reply update-charts-dep.sh]: added to package.sh

**Comment by elnyry** [comment]
@rmothilal can you please include central-ledger version v8.3.1 (I know we'll keep getting newer releases but would be nice to include it)

File: als-oracle-pathfinder/values.yaml [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+
+  pathfinder:
+    host: '156.154.59.224'
+    port: 7007
+    queryTimeoutMs: 10000
+    tls:
+      # rejectUnauthorized should be true in production
+      rejectUnauthorized: false
+      # The following paths are all relative to this directory containing the values file
+      client_cert_path: '/opt/als-oracle-pathfinder/secrets/client_cert.pem'
```
Comment (mdebarros) [inline als-oracle-pathfinder/values.yaml]: All these certs should be loaded from a secret(s), and then mounted into the correct location.

Comment (rmothilal) [reply als-oracle-pathfinder/values.yaml]: ok

Comment (rmothilal) [reply als-oracle-pathfinder/values.yaml]: https://github.com/mojaloop/helm/pull/218/files#diff-ba8c6ec03ed905aae6f9eebc36277168

**Comment by rmothilal** [comment]
This is a work in progress, the PR is not finalised yet

File: account-lookup-service/values.yaml [hunk 7]
```diff
@@ -289,10 +289,107 @@ als-oracle-pathfinder:
     log_level: 'info'
 
     pathfinder:
-      host: '156.154.59.224'
+      host: '10.0.0.1'
```
Comment (mdebarros) [inline account-lookup-service/values.yaml]: I would suggest changing this to something like `localhost`.

**Comment by rmothilal** [comment]
Screenshot 2019-11-11 at 15 59 42

**Comment by rmothilal** [comment]
Latests updates:
 ml-api to 8.4.1
 cl to 8.4.0
 cep to 8.4.0

**Pull request merged**